pmd-core/src/test/java/net/sourceforge/pmd/lang/document/TextFilesTest.java

```diff
@@ -0,0 +1,239 @@
+/*
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.document;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.assertTrue;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import org.apache.commons.io.IOUtils;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import net.sourceforge.pmd.PMDConfiguration;
+import net.sourceforge.pmd.lang.DummyLanguageModule;
+import net.sourceforge.pmd.lang.LanguageVersion;
+import net.sourceforge.pmd.util.datasource.DataSource;
+import net.sourceforge.pmd.util.datasource.FileDataSource;
+
+public class TextFilesTest {
+
+    @Rule
+    public TemporaryFolder tempDir = TemporaryFolder.builder().build();
+
+    private LanguageVersion dummyVersion = DummyLanguageModule.getInstance().getDefaultVersion();
+
+    @Test
+    public void testNioFile() throws IOException {
+        Path file = makeTmpFile(StandardCharsets.UTF_8, "some content");
+        try (TextFile tf = TextFile.forPath(file, StandardCharsets.UTF_8, dummyVersion)) {
+            assertEquals(file.toAbsolutePath().toString(), tf.getPathId());
+            assertEquals(file.toString(), tf.getDisplayName());
+            assertEquals(dummyVersion, tf.getLanguageVersion());
+            assertEquals(Chars.wrap("some content"), tf.readContents().getNormalizedText());
+        }
+    }
+
+    @Test
+    public void testEquals() throws IOException {
+        Path file = makeTmpFile(StandardCharsets.UTF_8, "some content").toAbsolutePath();
+        try (TextFile tf = TextFile.forPath(file, StandardCharsets.UTF_8, dummyVersion)) {
+            try (TextFile tf2 = TextFile.forCharSeq("some content", file.toString(), dummyVersion)) {
+                assertEquals(tf.getPathId(), tf2.getPathId());
+
+                assertNotEquals(tf, tf2);
+                assertNotEquals(tf2, tf);
+                assertEquals(tf, tf);
```

Comment:
That's unnecessary. Should we test for the display name maybe?

Reply:
I want to test the equals method... but `tf == tf` so indeed this is useless. I think we need another "tf" instance created to be equal but a different instance